### ISSUE TITLE
[Helm] Enable leader election by default for Fleet Managed agent installation with perNode preset

### ISSUE THREAD

**Issue by eedugon** [elastic/elastic-agent#6284]
When following the instructions at https://www.elastic.co/guide/en/fleet/current/example-kubernetes-fleet-managed-agent-helm.html and installing a Fleet Managed with preset `perNode` with something like:

```
helm install demo ./deploy/helm/elastic-agent \
--set agent.fleet.enabled=true \
--set agent.fleet.url=https://fleet-svc.default.svc \
--set agent.fleet.token=myToken \
--set agent.fleet.preset=perNode \
--set agent.fleet.insecure=true
```

We get the following `agent.yml` generated in a secret:

```
  agent.yml: |-
    fleet:
      enabled: true
    providers:
      kubernetes:
        node: ${NODE_NAME}
        scope: node
      kubernetes_leaderelection:
        enabled: false
        leader_lease: demo-pernode
```

If the helm chart disables the leader election then we shouldn't continue with the documentation instructions as the `kubernetes integration` default settings are intended for agents with the `leader` mechanism enabled.

I suggest to keep leader election enabled for this use case: fleet.enabled=true and fleet.preset=perNode

Of course the user could disable leaderElection if not needed / wanted.

cc: @pkoutsovasilis

**Comment by strawgate** [comment]
In this case, do we need to touch the providers configuration at all?

```
    providers:
      kubernetes:
        node: ${NODE_NAME}
        scope: node
      kubernetes_leaderelection:
        enabled: false
        leader_lease: demo-pernode
```

Wouldn't leader election be enabled by the Kubernetes integration if it was setup?

**Comment by eedugon** [comment]
The kubernetes integration is not capable of enabling / disabling the
provider or leader election mechanism, neither to influence the name of the
lease used for leader election. All that is configured at agent level
regardless of the integrations used.

I wish it could, it’s something I have said a few times it would be ideal
in my opinion, but probably not straight forward.



Edu González de la Herrán
Senior Technical Writer @ Elastic

*Please note: Emails may be stored and shared to provide a better customer
experience. To learn more about how we use your information, please review
Elastic’s **Privacy Statement. *
<https://www.elastic.co/legal/privacy-statement>

**Comment by strawgate** [comment]
Is the part about provider enablement still true with https://github.com/elastic/elastic-agent/pull/6169 having been merged?

**Comment by eedugon** [comment]
Oh! That’s awesome! I didn’t know about that change. Thanks!

When that’s implemented there will be a few things to consider, definitely.

Anyway, the scope of this issue is to confirm that when the user is
planning to use the kubernetes integration in a DS (per node)**for
kubernetes monitoring**, to ensure by default that the leader election is
enabled and with the expected lease name.

That will align with the integration policy default configuration of
retrieving almost everything (including cluster level events and metrics).

I don’t know if the PR you mentioned also influences the leader election
and lease name or just the provider.

Edu González de la Herrán
Senior Technical Writer @ Elastic

*Please note: Emails may be stored and shared to provide a better customer
experience. To learn more about how we use your information, please review
Elastic’s **Privacy Statement. *
<https://www.elastic.co/legal/privacy-statement>

**Comment by pkoutsovasilis** [comment]
@blakerouse please help us here; With this [PR](https://github.com/elastic/elastic-agent/pull/6169) merged does it mean that even if we pass at the agent a providers config like the one below
```
...
providers:
  kubernetes_leaderelection:
    enabled: true
...
```
when we don't have an input with a condition like the one below
```
condition: '${kubernetes_leaderelection.leader} == true'
```
the agent won't try to acquire a lease?

**Comment by pkoutsovasilis** [comment]
ok I tested this out and I can confirm that a lease is created only when there `${kubernetes_leaderelection.leader}` in the agent config. Amazing stuff 🙂 Then this is already resolved for 8.17.x agent versions?
